Sources/UtiliKit/ScrollingPageControl/ScrollingPageControl.swift

```diff
@@ -0,0 +1,411 @@
+//
+//  ScrollingPageControl.swift
+//  UtiliKit-iOS
+//
+//  Created by Nathan Chiu on 7/3/19.
+//  Copyright © 2019 Bottle Rocket Studios. All rights reserved.
+//
+
+import UIKit
+
+// MARK: -
+@IBDesignable
+class ScrollingPageControl: UIView {
+    
+    // MARK: Properties
+    
+    // MARK: Modeled Off UIPageControl
+    
+    /// The total number of pages represented, Default is `0`
+    @IBInspectable var numberOfPages: Int {
+        set { _numberOfPages = max(0, newValue) } // prevent a negative number of pages
+        get { return _numberOfPages }
+    }
+    
+    private var _numberOfPages: Int = 0 {
+        didSet {
+            currentPage = _currentPage // keep the current page in range
+            refreshDotLayout()
+        }
+    }
+    
+    /// The index of the currently selected page. Default is `0`. Value pinned to `0..numberOfPages-1`
+    @IBInspectable var currentPage: Int {
+        set {
+            _currentPage = max(0, min(newValue, numberOfPages - 1)) // clamp the newValue between 0 and numberOfPages
+        }
+        get { return _currentPage }
+    }
+    
+    private var _currentPage: Int = 0 {
+        didSet {
+            UIView.animate(withDuration: 0.3, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 0.0, options: .beginFromCurrentState, animations: { [weak self] in
```

Comment:
Why does this one not need to be weak? (just to better understand `weak self` personally) I was under the impression that anytime you call self in a block, you should default to making it weak to avoid memory leaks.

Reply:
You only need weak if the block is being retained. In this case `UIView.animate` doesn't retain the block, so you're free to use a strong self.

Reply:
Always better to be safe than sorry, but as a general rule of thumb, I'll usually omit `weak self` on `UIView` animation blocks as well as `DispatchQueue.main.async` blocks.